Sources/App/Core/PackageContributorService.swift

```diff
@@ -0,0 +1,133 @@
+            return try parseVCHistory(log: commitHistory)
+        } catch {
+            throw AppError.analysisError(packageID, "loadContributorsHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Gets the version control history in a string log
+    private static func queryVCHistory(cacheDirPath: String, packageID: UUID?) throws -> String {
+        
+        if !Current.fileManager.fileExists(atPath: cacheDirPath) {
+            throw AppError.unexistentPackageCacheDir(packageID, cacheDirPath)
+        }
+
+        // attempt to shortlog
+        do {
+            return try Current.git.shortlog(cacheDirPath)
+        } catch {
+            throw AppError.shellCommandFailed("gitShortlog",
+                                              cacheDirPath,
+                                              "queryVCHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Parses the result of queryVCHistory into a collection of contributors
+    private static func parseVCHistory(log: String) throws -> [Contributor] {
+        var committers = [Contributor]()
+        
+        for line in log.components(separatedBy: .newlines) {
+            let log = line.split(whereSeparator: { $0 == " " || $0 == "\t"})
+            
+            if (log.count != 2) {
+                var identifier = [String]()
+                for i in 1..<(log.count - 1) {
+                    identifier.append(String(log[i]))
+                }
+                
+                let committer = Contributor(numberOfCommits: Int(log.first!) ?? 0,
+                                            email: String(log.last!),
+                                            name: identifier.joined(separator: " "))
+                committers.append(committer)
+            }
+            
+        }
+        return committers
+    }
+    
+}
+
+
+
+/// Strategy for selecting contributors based entirely on the number of commits 
+struct CommitSelector {
+    
+    static func filter(candidates: [Contributor], threshold: Float) -> [Contributor] {
+        if candidates.isEmpty {
+            return []
+        }
+        
+        let maxNumberOfCommits = candidates.max(by: { (a,b) -> Bool in
+            return a.numberOfCommits < b.numberOfCommits
+        })!.numberOfCommits
+        
+        return candidates.filter { canditate in
+            return Float(canditate.numberOfCommits) > threshold * Float(maxNumberOfCommits)
+        }
+    }
+}
+
+
+struct PackageAuthors : Encodable, Decodable, Equatable {
+    var authors : [Author]
+    var numberOfContributors : Int 
+}
+
+
+
+final class PackageContributorService {
+    
+    
+    /// Extracts the possible authors of the package according to the number of commits.
+    /// A contributor is considered an author when the number of commits is at least a 60 percent
+    /// of the maximum commits done by a contributor
```

Comment:
I'm not sure I understand the comment, maths or functionality here.

We return and display an array of authors, but only one person can have ever created more than 60% of the total commits.

Or is the behaviour here that we take the top contributor (who is always an author), and then anybody who has more than 60% of that count is also an author? So essentially a top 40% performer?

(Not saying it's wrong but just some clarification on expected behaviour for my own brain aha)

Reply:
We take the top contributor (who is always an author), and then anybody who has *at least* 60% of that count is also an author. Say for instance the top contributor has 100 commits, then anybody who has at least 60 commits (a bit more than the half of that top count) is also an author. The 60% is actually arbitrary and rather irrelevant for our bigger purpose; the key point here is that we want to have a *default* way to extract authors information. There is no perfect way to get this right for all the cases. An algorithm that only takes into account the number of commits as a metric for authorship is certainly questionable. However, we do not need to dwell on this as the authors themselves will have the last word on authorship and what author information is shown 🤓. This last thing will be addressed in #174  :)

Reply:
Makes sense - thanks for explaining 😀

As you say hard to perfect as there are so many distributions of contributors (some repos are very flat with lots of contributions but all at the same level, but also some with basically one author with 90% of the total commits). 

The issue attached will probably resolve the "Core Maintainers" concept which some orgs institute.